docs/examples/101/custom-role-definition-assignment/main.bicep

```diff
@@ -0,0 +1,30 @@
+targetScope = 'subscription'
+
+param principalId string
+var roleName = 'Bicep Custom Role demo'
+
+resource definition 'Microsoft.Authorization/roleDefinitions@2018-01-01-preview' = {
+  name: guid(roleName)
+  properties: {
+    roleName: roleName
+    description: 'Custom role create with bicep'
+    permissions: [
+      {
+        actions: [
+          '*/read'
+        ]
+      }
+    ]
+    assignableScopes: [
+      subscription().id
+    ]
+  }
+}
+
+resource assignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(concat(roleName, principalId))

```

Comment:
instead of concatting these strings together, I would add arguments to the guid function (which can take as many arguments as you want to provide). I would also add the subscription id as an argument to ensure uniqueness:

```
name: guid(roleName, principalId, subscription().subscriptionId
```

Reply:
Yes, good idea ! 
I make the small tweak